// Controllers/ControllerUserInfo.js

var Session = require('Tools/Session');    

function Controller(){
    var request, response;
    Object.defineProperty(this, 'session', {value : new Session()});
}

Controller.prototype.Execute = function(entity, request, response){
    var store, sessionValue = [];
    this.request = request;
    this.response = response;
    
    store = core.GetStore(this.GetName());
    
    if (store == undefined){
        this.Send({error:"FATAL ERROR"});
        return;
    }
    
    this.session.Init(this.request, this.response);
    sessionValue = this.session.Get();

    if (sessionValue['sid'] == undefined){
        this.Send({});
        return;
    }
    
    this.Send({
		              profile_link: "profile link string",
		              name: "name string",
		              avatar: "avatar url string"
                    });
}

Controller.prototype.Send = function(result){
    this.response.writeHead(200, {"Content-Type": "application/json"});
    this.response.write(JSON.stringify({controller: "autorisation", entity: result}));
    this.response.end();
    
}

Controller.prototype.GetName = function(){
    return 'user_info';
}

module.exports = Controller;
