// Tools/Session.js

function Session() {
    var request, response, isEmpti, result;
};

Session.prototype.Init = function (request, response){
    this.isEmpti = false;
    this.request = request;
    this.response = response;
    this.result = [];
}

Session.prototype.Get = function(){
    var key = '', value = '', keyAdd = true;
    
    if (this.isEmpti) return this.result;
    
    if (this.request.headers.cookie == undefined) return [];
    
    for (i = 0; i < this.request.headers.cookie.length; i++){
        if (this.request.headers.cookie[i] == '='){keyAdd = false; continue;}
    
        if (this.request.headers.cookie[i] == ';'){
            this.result[key] = value;
            key = ''; value = '';
            keyAdd = true;
            if (i+1 < this.request.headers.cookie.length && this.request.headers.cookie[i+1] == ' ') i++;
            continue;
        }
    
        if (keyAdd){
            key += this.request.headers.cookie[i];
        }
        else{
            value += this.request.headers.cookie[i];
        }
    }
    if (key != '' && value != '') this.result[key] = value;
    this.isEmpti = true;
    return this.result;
};

Session.prototype.Set = function(key, value, expiresSession){
    if (expiresSession != undefined){
        expiresSession = '';
    }
    else{
        expiresSession = 'expires=Mon, 28-Dec-2020 13:46:00 GMT;';
    }
    this.result[key] = value;
    this.response.setHeader("Set-Cookie", key + '='+ value + '; path=/; ' + expiresSession);
    return true;
};

module.exports = Session;
