// StorePair/Memcache.js

// TODO : модуль не дописан
var memcache = require('Tools/Memcache');

Memcache.prototype=new Object();
Memcache.prototype.constructor=Memcache;

function Memcache(port, host){
    this.port = port || 11211;
    this.host = host || 'localhost';
    this.client = new memcache.Client(this.port, this.host);
    
    this.client.on('connect', function(){
        this.connect = true;
    });
    
    this.client.on('close', function(){
        this.conect = false;
    });
    
    this.client.connect();
    
    this.client.on('error', function(e){
        
    }
}

Memcache.prototype.Get = function (key){
   this.client.get(key, function(error, result){
       
    });
}

Memcache.prototype.Set = function (key, value){
    this.client.set(key, value, function(error, result){
        
    });
}

Memcache.prototype.Delete = function (key){
    this.client.delete(key, function(error, result){
       
    });
}

Memcache.prototype.Destroy = function (){
    this.client.close();
    delete this.client();
}
module.exports = Memcache;
